code/data_structures/src/queue/queue_using_stack/queue_using_stack.c

```diff
@@ -0,0 +1,108 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+struct stack 
+{
+    int top;
+    unsigned capacity;
+    int* A;
+};
+
+struct queue
+{
+    struct stack* S1;
+    struct stack* S2;
+};
+
+//STACK
+struct stack* createStack(unsigned capacity) 
+{
+    struct stack* newStack = (struct stack*)malloc(sizeof(struct stack));
+    newStack->capacity = capacity;
+    newStack->top = -1;
+    newStack->A = (int*)malloc(capacity*sizeof(int));
+    return newStack;
+}
+
+int isFull(struct stack* S)
+{
+    return (S->top == S->capacity - 1);
+}
+
+int isEmpty(struct stack* S)
+{
+    return (S->top == -1);
+}
+
+void push(struct stack* S, int item)
+{
+    if (!isFull(S))
+    {
+        S->A[++S->top] = item;
+    }
+    return;
+}
+
+int pop(struct stack* S)
+{
+    if (!isEmpty(S))
+    {
+        return (S->A[S->top--]);
+    }
```

Comment:
Please return a default value  to avoid UB.

Reply:
Will use of assertion be fine? If not, what default value should I use?

Reply:
Sure, you could use an assert.

Reply:
Done. Anyways, I would like to know what could have been used as default value here.